Plan/fabric/build.gradle

```diff
@@ -0,0 +1,63 @@
+plugins {
+    id 'fabric-loom' version '0.8-SNAPSHOT'
+}
+
+dependencies {
+    shadow "net.playeranalytics:platform-abstraction-layer-api:$palVersion"
+    implementation project(path: ":common", configuration: 'shadow')
+    shadow project(path: ":common", configuration: 'shadow')
+    compileOnly project(":api")
+    modImplementation('me.lucko:fabric-permissions-api:0.1-SNAPSHOT')
+
+    minecraft "com.mojang:minecraft:1.17.1"
+    mappings "net.fabricmc:yarn:1.17.1+build.21:v2"
+    modImplementation "net.fabricmc:fabric-loader:0.11.6"
+
+    // Fabric API
+    Set<String> apiModules = [
+            'fabric-api-base',
+            'fabric-command-api-v1',
+            'fabric-entity-events-v1',
+            'fabric-lifecycle-events-v1',
+            'fabric-networking-api-v1'
+    ]
+
+    apiModules.forEach {
+        modImplementation(fabricApi.module(it, "0.37.0+1.17"))
+    }
+
+    testImplementation project(path: ":common", configuration: 'testArtifacts')
+}
+
+compileJava {
+    options.release = 16
```

Comment:
This probably needs to be defined as 8 in main build.gradle for submodules so that other modules build with 8 even with newer JDK versions.

Could you also change java version of pipelines in .github to 16 so that the tests can run - I'll do rest of the review a bit later when I have more time